### PR TITLE
refactor, chore(egen): refactors code as per new template guidelines, removes future tense, replaces REGION with LOCATION, removes generate_uuid function

### PR DESCRIPTION
**Pull request by sumanvita-springml**
This PR includes the following changes:

1. changes REGION to LOCATION
2. Removes function that generates UUID
3. Includes colab restart kernal and authentication as per template guidelines
4. Adds collab enterprise logo
5. Makes structural changes according to the new notebook template
6. Removes the use of future tense
7. Contraction of words like you are to you're, it is to it's
8. Adds code highlight

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: notebooks/official/workbench/pricing_optimization/pricing-optimization.ipynb [hunk 1]
```diff
@@ -402,7 +319,7 @@
         "id": "3a063f530682"
       },
       "source": [
-        "If you are using ***Vertex AI Workbench managed notebooks instance***, every cell which starts with \"#@bigquery\" will be a SQL Query. If you are using Vertex AI Workbench user managed notebooks instance or Colab it will be a markdown cell."
+        "If you're using ***Vertex AI Workbench managed notebooks instance***, every cell which starts with \"#@bigquery\" is a SQL Query. If you're using a Vertex AI Workbench user-managed notebook instance or Colab, it's a markdown cell."
```
Comment (kittyabs) [inline notebooks/official/workbench/pricing_optimization/pricing-optimization.ipynb:322]: change to: "Vertex AI Workbench user-managed notebooks instance" (plural on "notebooks")

**Review by kittyabs — COMMENTED** [review]
edit

**Review by katiemn — APPROVED** [review]

**Pull request merged**